cleangoldenami/main.go

```diff
@@ -115,26 +183,69 @@ func getPublicImages(ec2Client DescribeImagesClient) ([]ec2Types.Image, error) {
 	return describeImagesResponse.Images, nil
 }
 
-func getOldestImage(images []ec2Types.Image) (ec2Types.Image, error) {
+func hasMatchingTag(image ec2Types.Image, tag string) bool {
+	for _, t := range image.Tags {
+		if *t.Key == "version" && *t.Value == tag {
+			return true
+		}
+	}
+	return false
+}
+
+func hasMatchingArchitecture(image ec2Types.Image, architecture string) bool {
+	return string(image.Architecture) == architecture
```

Comment:
instead of wrapping this in a `string`, let's make `architecture` here into type `ec2Types.ArchitectureValues` all the way up the call stack. Then on lines 70-72 you can use `ec2Types.ArchitectureValuesArm64` and so on.